internal/resources/provisioner/data_source_provisioner.go

```diff
@@ -83,12 +91,38 @@ func dataSourceProvisionerRead(ctx context.Context, d *schema.ResourceData, m in
 			return
 		}
 
-		d.SetId(resp.Provisioner.Meta.UID)
-
-		if err := d.Set(common.MetaKey, common.FlattenMeta(resp.Provisioner.Meta)); err != nil {
-			return diag.FromErr(err)
+		p := &provisioner.VmwareTanzuManageV1alpha1ManagementclusterProvisionerListprovisionersResponse{
+			Provisioners: []*provisioner.VmwareTanzuManageV1alpha1ManagementclusterProvisionerProvisioner{
+				{
+					FullName: &provisioner.VmwareTanzuManageV1alpha1ManagementclusterProvisionerFullName{
+						ManagementClusterName: getResp.Provisioner.FullName.ManagementClusterName,
+						Name:                  getResp.Provisioner.FullName.Name,
+						OrgID:                 getResp.Provisioner.FullName.OrgID,
+					},
+					Meta: &objectmetamodel.VmwareTanzuCoreV1alpha1ObjectMeta{
+						Description:     getResp.Provisioner.Meta.Description,
+						Labels:          getResp.Provisioner.Meta.Labels,
+						UID:             getResp.Provisioner.Meta.UID,
+						ResourceVersion: getResp.Provisioner.Meta.ResourceVersion,
+					},
+				},
+			},
 		}
+
+		resp = p
+	}
+
+	err = tfModelDataConverter.FillTFSchema(resp, d)
+
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to populate tf schema"))
 	}
 
+	for _, prov := range resp.Provisioners {
```

Comment:
With this logic , If you invoke the datasource multiple times can it create multiple entries in the state file? because
1.  the response might not maintain the order of provisioners 
2. A new provisioner is added then id will definitely change with this new field. Can you test this part?

Reply:
On invoking datasource multiple times, if there is a change in the provisioners state from TMC server side (that is a new create/delete/update), the unique ID changes but it does not create multiple entries. I have tested the same